content/200-concepts/100-components/01-prisma-schema/06-relations/420-referential-integrity.mdx

```diff
@@ -35,37 +67,43 @@ generator js {
+
+- `foreignKeys`: this handles referential integrity in the database using foreign keys. This is the default option when using all SQL database connectors.
+- `prisma`: this emulates referential integrity in the Prisma client. This is the default (and only available) option when using the MongoDB connector.
+
+<Admonition type="warning">
+  Setting the value to `prisma` will remove all previously created foreign keys
+  in the next migration when using Prisma Migrate. Setting the value to
+  `foreignKeys` will create foreign keys for all relations in the next migration
+  when using Prisma Migrate.
+</Admonition>
 
 ## Handling the referential integrity in Prisma
 
-If the integrity is handled by the Prisma Client, for now there will be reduced amount of available [referential actions](/concepts/components/prisma-schema/relations/referential-actions) available in the data model:
+If referential integrity is handled by the Prisma Client, there is currently a reduced number of supported [referential actions](/concepts/components/prisma-schema/relations/referential-actions) available in the data model:
 
-| action     | `onUpdate` | `onDelete` |
-|:-----------|:-----------|:-----------|
-| Restrict   | no         | yes        |
-| SetNull    | no         | yes        |
-| SetDefault | no         | no         |
-| Cascade    | no         | yes        |
-| NoAction   | yes        | yes        |
+| Referential action | `onUpdate` | `onDelete` |
+| :----------------- | :--------- | :--------- |
+| Restrict           | no         | yes        |
+| SetNull            | no         | yes        |
+| SetDefault         | no         | no         |
+| Cascade            | no         | yes        |
+| NoAction           | no         | yes        |
 
 In this mode, any [`raw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) database queries will not trigger any of the actions, and they should be handled manually if deleting or updating records using the [`raw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) queries.
 
-The relations are needed for the client to be able to join data between models, so when using the `db pull` command to a database with the `prisma` integrity, existing relations are kept in the data model. This has some caveats:
+The relations are needed for the client to be able to join data between models, so when using the `db pull` command to a database with `referentialIntegrity` set to `prisma` in the `datasource`, existing relations are kept in the data model. This has some caveats:
 
 - When a model in the database is deleted, the relation on the other side gets deleted, if existing.
 - When a scalar field used in a relation field gets deleted or renamed, the relation field is kept as-is, leading to validation errors. The user must fix the relation before continuing to use the data model in this case.
```

Comment:
I am not sure if I understand what this means.

Reply:
Not sure I do either!

Reply:
@pimeys would this be something you're able to clarify?

Reply:
Yeah this is badly written. What it means is when we have a situation like this:

```prisma
model A {
  field Int
  b     B   @relation(fields: [field], references: [id])
}
```
You then modify your database, change the `field` name to `renamedField`, which leads to the following data model:
```prisma
model A {
  renamedField Int
  b            B   @relation(fields: [field], references: [id])
}
```
As you can see the `fields` parameter points to a non-existing field in the model, and this is something we can't fix in the re-introspection: the user must do it manually.

Reply:
Updated with this example